src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const LENGTH = 'length';
+const ALPHABET = 'alphabet';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (goods, { sortedField, reversed }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortedField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortedField) {
+        case ALPHABET:
+          return good1.localeCompare(good2);
+        case LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
```

Comment:
[Not critical, not required]
This approach works fine, but it's possible to make sorting even faster: handle `reversed` when you call `sort()` method. Way of sorting would be the same, just the order of parameters would change.
For example:
```
const sortGoods = (good1, good2) => {
  switch (sortedField) {
    case ALPHABET:
      return good1.localeCompare(good2);
    case LENGTH:
      return good1.length - good2.length;
    default:
      return 0;
  }
};

if (reversed) {
  preparedGoods.sort((good1, good2) => sortGoods(good2, good1))
} else {
  preparedGoods.sort(sortGoods);
}
```